doc/api/worker_threads.md

```diff
@@ -1527,6 +1527,79 @@ Calling `unref()` on a worker allows the thread to exit if this is the only
 active handle in the event system. If the worker is already `unref()`ed calling
 `unref()` again has no effect.
 
+## `worker.makeSync(buffer[, options])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `buffer` {SharedArrayBuffer} A shared memory buffer to use for communication.
+* `options` {Object}
+  * `timeout` {number} The timeout in milliseconds for synchronous calls. **Default:** `5000`.
+  * `expandable` {boolean} Whether the buffer can be resized. **Default:** `true` if the buffer
+    supports `growable` option.
+* Returns: {Object} An object with synchronous methods mirroring those exposed through [`worker.wire()`][].
+
+Creates a synchronous API facade that communicates with a worker thread over a shared memory buffer.
```

Comment:
We have the `receiveMessageOnPort()` API specifically so that we can have full-featured synchronous message passing. While this can be implemented through shared memory, I don't currently see a good reason to do so (beyond the flags used for `Atomics.wait()`/`Atomics.notify()` of course, but that could be fully internal).

Reply:
Are you recommending implementing this as:

1. The caller sends a `postMessage()` down, with a new port and a new `SharedArrayBuffer`.
2. caller calls `Atomics.wait()` on said `SharedArrayBuffer`
3. The receiver normally calls a `postMessage()` on the passed in port. Then `Atomics.notify()`.
4. The caller calls `receiveMessageOnPort()` to get the response?

Can we guarantee that `receiveMessageOnPort ()` will always find a message in this case, and there is no race condition?

Reply:
> with a new port and a new `SharedArrayBuffer`.

The `SharedArrayBuffer` could be the first message sent over the port, too, so you'd end up _only_ needing the `MessagePort` to pass to the other thread.

> Can we guarantee that `receiveMessageOnPort ()` will always find a message in this case, and there is no race condition?

This would come with the same guarantees as the current implementation. You obviously need to think about cases like "Two messages coming in from the receiver at the same time", but that's already the case. As long as you stick to standard multithreaded programming patterns, you're fine.